code/model.py

```diff
@@ -1099,6 +1152,24 @@ def forward(self, batch):
         return self.output
 
 
+    def get_weighted_values(self, output):
+        """
+        Calculations for values for polarity attention.
+        First few epochs guide attention values according to true labels for target and expression.
+        See imn/code/my_layers.py:Self_attention().call() for more. 
+        """
+        # PICK UP HERE
+        # Could help guide model in correct direction
+        # predicted prob of B or I in target and expression outputs
+        # bi_probs = (target_output[:,:,1:].sum(dim=-1) + expression_output[:,:,1:].sum(dim=-1))/2.
+        # bi_probs = bi_probs[:,:polarity_output.size(2)]
+        # values =  torch.mul(
+        #     polarity_output.permute(1, 0, 2), # embedding, batch, sequence
+        #     bi_probs,
+        # ).permute(2, 1, 0)  # sequence, batch, embedding
+        return None
```

Comment:
This helper function meant to create values based on how you want to train you polarity self attention.